src/mem/address_space.h

```diff
@@ -236,5 +236,21 @@ namespace snmalloc
 
       return res;
     }
+
+    /**
+     * Default constructor.  An address-space manager constructed in this way
+     * does not own any memory at the start and will request any that it needs
+     * from the PAL.
+     */
+    AddressSpaceManager() = default;
```

Comment:
What's the use case for this constructor? How can the AS manager manage the memory if it doesn't add a range?

The `Sandbox` class below reserves the memory in the constructor, I was expecting it to be passed down to the AS manager on its own construction.

Reply:
Precisely as the comment says: it requests memory from the PAL.  This is how it works normally in the non-sandbox use case.

Reply:
Right, then why do you also need the second c-tor, with pointer and size?

Reply:
For the sandbox use case, where the PAL is not able to allocate any memory and all of the memory that it has access to is provided in the constructor.

Reply:
Could the NO_ALLOCATION Pal feature be used to disable this constructor?

Reply:
Not very easily.  This ends up deleting a constructor that's used in a load of other places and propagates up into LargeAlloc and its `make` function.

Reply:
I see, one for the external alloc and one for the internal alloc. Makes sense, thanks!